From fffe20d84f9dec2c86a1eb6322e472172fdf9e73 Mon Sep 17 00:00:00 2001
From: bert hubert <bert.hubert@netherlabs.nl>
Date: Sun, 11 Sep 2016 20:40:44 +0200
Subject: [PATCH] eleksir noted that we leak a ton of memory in postgresql. I'm
 no postgres expert, but this plugs my leak and still appears to function. In
 other news, do we need a transaction for every query?

(cherry-picked from master 903bb4924bc5130c0e81f0c5759d0177f53e82fd)
---
 modules/gpgsqlbackend/spgsql.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules/gpgsqlbackend/spgsql.cc b/modules/gpgsqlbackend/spgsql.cc
index 860a2db..ed80d4c 100644
--- a/modules/gpgsqlbackend/spgsql.cc
+++ b/modules/gpgsqlbackend/spgsql.cc
@@ -77,7 +77,8 @@ class SPgSQLStatement: public SSqlStatement
       L<<Logger::Warning<<"Query: "<<d_query<<endl;
     }
     if (!d_parent->in_trx()) {
-      PQexec(d_db(),"BEGIN");
+      auto res=PQexec(d_db(),"BEGIN");
+      PQclear(res);
       d_do_commit = true;
     } else d_do_commit = false;
     d_res_set = PQexecPrepared(d_db(), d_stmt.c_str(), d_nparams, paramValues, paramLengths, NULL, 0);
